engine/cdn/cdn_log.go

```diff
@@ -0,0 +1,232 @@
+package cdn
+
+import (
+	"bufio"
+	"context"
+	"crypto/rsa"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	gocache "github.com/patrickmn/go-cache"
+
+	"github.com/ovh/cds/engine/api/services"
+	"github.com/ovh/cds/engine/api/worker"
+	"github.com/ovh/cds/engine/api/workflow"
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/jws"
+	"github.com/ovh/cds/sdk/log"
+	"github.com/ovh/cds/sdk/log/hook"
+)
+
+var (
+	logCache = gocache.New(20*time.Minute, 30*time.Minute)
+)
+
+func (s *Service) RunTcpLogServer(ctx context.Context) {
+	log.Info(ctx, "Starting tcp server %s:%d", s.Cfg.TCP.Addr, s.Cfg.TCP.Port)
+	listener, err := net.Listen("tcp", fmt.Sprintf("%s:%d", s.Cfg.TCP.Addr, s.Cfg.TCP.Port))
+	if err != nil {
+		log.Fatalf("unable to start tcp log server: %v", err)
+	}
+
+	//Gracefully shutdown the tcp server
+	go func() {
+		select {
+		case <-ctx.Done():
+			log.Info(ctx, "CDN> Shutdown tcp log Server")
+			_ = listener.Close()
+		}
+	}()
+
+	go func() {
```

Comment:
sdk.Goroutine ?

Reply:
I don’t think for this particular usecase, but we can talk about it